### PR TITLE
Feature/sep6

### PR DESCRIPTION
**Pull request by GonzaloAlt**
# Summary  

Implement and refactor missing SEP6 necessary endpoints for off/on-chain operations.  

* Add optional properties to the operations created above.
* Split responsibilities between SEP24 and SEP6 operations.  
* Update Anclap credentials.  
* Create interfaces and classes for requests & responses.  
* Create nodes for /fee, /info, /deposit-exchange, /withdraw-exchange.

# Evidence  

</img>
</img>
</img>

# References  

📖 [SEP-6 documentation](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0006.md)

### CODE REVIEW

File: nodes/Anclap/actions/info/fee/execute.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { IExecuteFunctions } from "n8n-core";
+import AnclapCredentials from "../../../transport/AnclapCredentials";
+import { OffChainOperationType, TransactionType } from "../../../transport/types";
+import SEP6 from '../../../transport/SEP6';
+import FeeRequest from "../../../transport/requests/FeeRequest/FeeRequest";
+
+export async function fee(this: IExecuteFunctions){
+    const anclapCredentials = new AnclapCredentials(await this.getCredentials('anclapApi'));
```
Comment (Genebson) [inline nodes/Anclap/actions/info/fee/execute.ts:8]: This line has an extra tab.

File: nodes/Anclap/actions/info/transaction/execute.ts [hunk 2]
```diff
@@ -3,26 +3,55 @@ import SEP6 from '../../../transport/SEP6';
...
+			stellarTransactionId,
+			externalTransactionId,
+			lang,
+		});
+	}else{
+		if(!id){
+			throw new Error('On of id or stellar Transaction Id or external Transaction Id is required');
+		}
+		transactionRequest = new TransactionRequest({id});
 	}
```
Comment (Genebson) [inline nodes/Anclap/actions/info/transaction/execute.ts:40]: This whole condition needs to be formatted.

File: nodes/Anclap/actions/info/transfer-server/execute.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { IExecuteFunctions } from "n8n-workflow";
+import AnclapCredentials from "../../../transport/AnclapCredentials";
+import SEP6 from "../../../transport/SEP6";
+import TransferServerRequest from "../../../transport/requests/TransferServerRequest/TransferServerRequest";
+
+export async function transferServer(this: IExecuteFunctions){
+    const anclapCredentials = new AnclapCredentials(await this.getCredentials('anclapApi'));
+
+    const getSep6TransferServerInfo = async () => {
+		const sep6 = new SEP6(anclapCredentials, '');
```
Comment (Genebson) [inline nodes/Anclap/actions/info/transfer-server/execute.ts:10]: This needs to be formatted too.

File: nodes/Anclap/actions/transactions/deposit-exchange/execute.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+    const sourceAsset = this.getNodeParameter('sourceAsset', 0) as string;
+    const destinationAsset = this.getNodeParameter('destinationAsset', 0) as string;
+    const amount = this.getNodeParameter('amount', 0) as string;
+    const type = this?.getNodeParameter('type', 0) as string;
+
+    const showOptionalValues = this.getNodeParameter('showOptionalValues', 0) as boolean;
+    
+    const getSep6DepositExchangeInfo = async () => {
+		const sep6 = new SEP6(anclapCredentials, token);
```
Comment (Genebson) [inline nodes/Anclap/actions/transactions/deposit-exchange/execute.ts:20]: Extra tab.

File: nodes/Anclap/actions/transactions/deposit-exchange/execute.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        }else{
+            depositExchangeRequest = new DepositExchangeRequest({
+                sourceAsset,
+                amount,
+                destinationAsset,
+                type
+            })
+        }
+        return await sep6.getDepositExchange(depositExchangeRequest);
+    }
```
Comment (Genebson) [inline nodes/Anclap/actions/transactions/deposit-exchange/execute.ts:58]: Needs formatting.

File: nodes/Anclap/actions/transactions/deposit-interactive/execute.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { IExecuteFunctions } from 'n8n-workflow';
+import AnclapCredentials from '../../../transport/AnclapCredentials';
+import { AnclapAssetCode } from '../../../transport/responses/IAnclapTransferServerInfoResponse';
+import SEP24 from '../../../transport/SEP24';
+
+export async function depositInteractive(this: IExecuteFunctions){
+    const anclapCredentials = new AnclapCredentials(await this.getCredentials('anclapApi'));
+	const token = this.getNodeParameter('token', 0) as string;
+	const assetCode = this.getNodeParameter('assetCode', 0) as AnclapAssetCode;
```
Comment (Genebson) [inline nodes/Anclap/actions/transactions/deposit-interactive/execute.ts:9]: Extra tab.

File: nodes/Anclap/actions/transactions/deposit-interactive/execute.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+    const getSep24DepositUrl = async () => {
+		const sep24 = new SEP24(anclapCredentials, token);
+		return await sep24.getDepositInteractiveUrl(assetCode);
```
Comment (Genebson) [inline nodes/Anclap/actions/transactions/deposit-interactive/execute.ts:13]: Extra tab.

File: nodes/Anclap/actions/info/fee/execute.ts (hunk 1 above)
Comment (GonzaloAlt) [reply nodes/Anclap/actions/info/fee/execute.ts]: Solved, the code has been formatted.

File: nodes/Anclap/actions/info/transaction/execute.ts (hunk 2 above)
Comment (GonzaloAlt) [reply nodes/Anclap/actions/info/transaction/execute.ts]: Solved, the code has been formatted

File: nodes/Anclap/actions/transactions/deposit-exchange/execute.ts (hunk 4 above)
Comment (GonzaloAlt) [reply nodes/Anclap/actions/transactions/deposit-exchange/execute.ts:19]: Solved, the code has been formatted

File: nodes/Anclap/actions/transactions/deposit-exchange/execute.ts (hunk 5 above)
Comment (GonzaloAlt) [reply nodes/Anclap/actions/transactions/deposit-exchange/execute.ts]: Solved, the code has been formatted.

**Pull request merged**